src/helpers.rs

```diff
@@ -463,18 +463,6 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
         }
     }
 
-    /// Dispatches to appropriate implementations for reading an OsString from Memory,
-    /// depending on the interpretation target.
-    /// FIXME: Use `Cow` to avoid copies
-    fn read_os_str_from_target_str(&self, scalar: Scalar<Tag>) -> InterpResult<'tcx, OsString> {
```

Comment:
Why remove this one but not the `write` and `alloc` variant?

Reply:
Okay the `alloc` variant still seems used (we should get rid of it, but not in this PR). But what about the `write` variant?

Reply:
> we should get rid of it, but not in this PR

Well actually if you want you can add that to this PR as well: remove `alloc_os_str_as_target_str` and `alloc_env_var_as_target_str`. Instead the loop in `EnvVars::init` can do the `match` itself.